Presentation.Web/app/services/exportGridToExcelService.ts

```diff
@@ -13,54 +14,59 @@
             if (!this.exportFlag) {
                 e.preventDefault();
                 _.forEach(columns, column => {
-                    if (column.hidden) {
-                        column.tempVisual = true;
-                        e.sender.showColumn(column);
-                    }
+                    //check if column has any related columns
+                    var columnsWithSameFieldName =  columns.filter(x => x.field === column.field);
```

Comment:
@a-naskret I think this approach will bite us in the a** :-)

We want to be explicit about the relationship of columns because dependency on the same field name is not the same as one being an "excel only column" and the other being a real one..

Check this one: 
![image](https://user-images.githubusercontent.com/13116889/164010122-e35cc562-4b18-431c-a3d9-9f0dbef4fd7c.png)

The two columns are related .. not by the dependency on the data field but by the logical relationship in the excel output.

I recommend extending the "excel builder" with the Id of the "controlling/master" column and if you find a column which has a "master column" , check if that one is visible or not. Then it is no longer "by chance"/accident, but very much visible that the columns are related.

Reply:
Also consider reordering the "depender" on the rhs of the "master"